src/common/health.rs

```diff
@@ -0,0 +1,104 @@
+use std::collections::HashSet;
+use std::ops::DerefMut as _;
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use collection::shards::replica_set::ReplicaState;
+use collection::shards::shard::ShardId;
+use collection::shards::CollectionId;
+use storage::content_manager::toc::TableOfContent;
+use tokio::sync::RwLock;
+
+pub struct Ready {
+    toc: Arc<TableOfContent>,
+    consensus_sync_instant: Instant,
+    initial_unhelathy_shards: RwLock<Option<HashSet<Replica>>>,
+}
+
+impl Ready {
+    pub fn new(toc: Arc<TableOfContent>) -> Self {
+        Self::with_consensus_sync_timeout(toc, Duration::from_secs(10))
+    }
+
+    pub fn with_consensus_sync_timeout(
+        toc: Arc<TableOfContent>,
+        consensus_sync_timeout: Duration,
+    ) -> Self {
+        Self {
+            toc,
+            consensus_sync_instant: Instant::now() + consensus_sync_timeout,
+            initial_unhelathy_shards: Default::default(),
+        }
+    }
+
+    pub async fn ready(&self) -> bool {
+        if Instant::now() < self.consensus_sync_instant {
+            return false;
+        }
```

Comment:
The only reasonable way to handle consensus synchronization on startup (that I've come up with) is to _always_ return "not ready" for the first `n` seconds after startup (to give the node some time to catch up with consensus).